staker-rewards/test/sla-calculator.test.js

```diff
@@ -0,0 +1,134 @@
+import chai from "chai"
+import testCache from "./data/test-cache.json"
+import SLACalculator from "../lib/sla-calculator.js"
+
+const { assert } = chai
+
+const operator = "0xF1De9490Bf7298b5F350cE74332Ad7cf8d5cB181"
+
+const mockCache = {
+    getKeeps: (status) =>
+        testCache.keeps.filter(keep => !status || keep.status.name === status)
+}
+
+describe("SLA calculator", () => {
+    it('should return keygen SLA equal to 100% ' +
+        'if operator has no failed keygens', () => {
+        // The operator has `2` keeps created within this interval. Both keeps
+        // are currently active. This means `2` key generations occurred and
+        // all of them completed successfully.
+        const interval = {
+            start: 9000,
+            end: 20000
+        }
+
+        const slaCalculator = SLACalculator.initialize(mockCache, interval)
+
+        const operatorSLA = slaCalculator.calculateOperatorSLA(operator)
+
+        assert.equal(operatorSLA.keygenCount, 2)
+        assert.equal(operatorSLA.keygenFailCount, 0)
+        assert.equal(operatorSLA.keygenSLA, 100)
+    })
```

Comment:
Can we add a test covering 100% key-gen SLA when one keep is already closed or terminated? It should be possible if we set the interval start at `8000`.

Reply:
Ah, my bad - I see now this test covers the scenario I mentioned. I misread `20000` vs `2000` and though we are just taking the first two keeps from the cache.